### PR TITLE
Enable ARM64.

### PR DESCRIPTION
**Pull request by JunielKatarn**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [x] I am making a change required for Microsoft usage of react-native

#### Description of changes

Adds macros to enable ARM64 compilation.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native/pull/113)

### CODE REVIEW

**Comment by acoates-ms** [comment]
Let's make sure to get this submitted to Facebook/folly too.  I am trying to get rid of the folly fork altogether.

File: Folly/folly/portability/Builtins.h [hunk 1]
```diff
@@ -74,7 +74,7 @@ FOLLY_ALWAYS_INLINE int __builtin_ctzl(unsigned long x) {
   return __builtin_ctz((unsigned int)x);
 }
 
-#if defined(_M_IX86) || defined(_M_ARM)
+#if defined(_M_IX86) || defined(_M_ARM) || defined(_M_ARM64)
```
Comment (NikoAri) [inline Folly/folly/portability/Builtins.h]: >#if defined [](start = 0, length = 11)

Could you add a quick comment of why these differences are needed between platforms?

File: Folly/folly/portability/Builtins.h [hunk 2]
```diff
@@ -100,7 +100,7 @@ FOLLY_ALWAYS_INLINE int __builtin_ffsl(long x) {
   return __builtin_ffs(int(x));
 }
 
-#if defined(_M_IX86) || defined(_M_ARM)
+#if defined(_M_IX86) || defined(_M_ARM) || defined(_M_ARM64) // Non-amd64 platforms
```
Comment (aeulitz) [inline Folly/folly/portability/Builtins.h]: >// Non-amd64 platforms [](start = 61, length = 22)

FWIW, if the intention is as the comment says, could this be abbreviated to '#ifndef _M_X64' or '#ifndef _M_AMD64'?

Comment (JunielKatarn) [reply Folly/folly/portability/Builtins.h]: Maybe, but that's assuming we know all possible _M_* possibilities.
This one is less disruptive and more likely to be accepted by Facebook (parallel PR already sent).

**Pull request merged**